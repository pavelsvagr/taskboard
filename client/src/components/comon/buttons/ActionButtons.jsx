import { Affix } from "antd"
import React from "react"
import PropTypes from "prop-types"

function ActionButtons({ children }) {
  return (
    <Affix
      offsetBottom={30}
      style={{ textAlign: "right" }}
      className="button-action" 
    >
      {children}
    </Affix>
  )
}

ActionButtons.propTypes = {
  children: PropTypes.node.isRequired
}

export default ActionButtons
