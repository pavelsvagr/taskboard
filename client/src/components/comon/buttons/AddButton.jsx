import React from "react"
import { PlusOutlined } from "@ant-design/icons"
import PropTypes from "prop-types"
import ResizableButton from "./ResizableButton"

function AddButton({ onClick, type = "primary", ...otherProps }) {
  return (
    <ResizableButton
      onClick={onClick}
      type={type}
      icon={<PlusOutlined />}
      {...otherProps}
    />
  )
}

AddButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string,
}

AddButton.defaultProps = {
  type: "primary",
}

export default AddButton
