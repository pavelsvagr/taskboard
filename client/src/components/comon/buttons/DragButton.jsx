import React from "react"
import { Button } from "antd"
import { MenuOutlined } from "@ant-design/icons"
import PropTypes from "prop-types"

function DragButton({ provider, className, size }) {
  return (
    <Button
      className={className}
      size={size}
      shape="circle"
      ghost
      icon={<MenuOutlined />}
      {...provider.dragHandleProps}
    />
  )
}

DragButton.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  provider: PropTypes.object.isRequired,
  className: PropTypes.string,
  size: PropTypes.string,
}

DragButton.defaultProps = {
  className: null,
  size: null,
}

export default DragButton
