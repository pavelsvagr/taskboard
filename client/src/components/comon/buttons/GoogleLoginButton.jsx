import React from "react"
import { GoogleOutlined } from "@ant-design/icons"
import { Button } from "antd"

function GoogleLoginButton() {
  return (
    <Button
      className="login"
      href="/auth/google"
      type="primary"
      icon={<GoogleOutlined />}
      size="large"
    >
      Login With Google
    </Button>
  )
}

export default GoogleLoginButton