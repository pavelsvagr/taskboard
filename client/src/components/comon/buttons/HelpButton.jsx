import React from "react"
import { QuestionOutlined } from "@ant-design/icons"
import { Button } from "antd"
import PropTypes from "prop-types"

function HelpButton({ size, onClick }) {
  return (
    <Button
      shape="circle"
      type='dashed'
      size={size}
      icon={<QuestionOutlined />}
      onClick={onClick}
    />
  )
}

HelpButton.propTypes = {
  size: PropTypes.string,
  onClick: PropTypes.func.isRequired
}

HelpButton.defaultProps = {
  size: "large"
}

export default HelpButton