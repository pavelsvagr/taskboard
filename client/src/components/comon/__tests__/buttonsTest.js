/* eslint react/jsx-filename-extension:0 */
import React from "react"
import renderer from "react-test-renderer"
import windowMedia from "../../../helpers/tests/windowMedia"
import AddButton from "../buttons/AddButton"
import DragButton from "../buttons/DragButton"
import GoogleLoginButton from "../buttons/GoogleLoginButton"
import HelpButton from "../buttons/HelpButton"
import ActionButtons from "../buttons/ActionButtons"

describe("Buttons", () => {
  it("Snapshot: AddButton.jsx", () => {
    Object.defineProperty(window, "matchMedia", windowMedia)
    const tree = renderer.create(<AddButton onClick={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Snapshot: DragButton.jsx", () => {
    Object.defineProperty(window, "matchMedia", windowMedia)
    const tree = renderer
      .create(
        <DragButton provider={{ dragHandleProps: {} }} onClick={jest.fn()} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Snapshot: GoogleLoginButton.jsx", () => {
    Object.defineProperty(window, "matchMedia", windowMedia)
    const tree = renderer.create(<GoogleLoginButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Snapshot: HelpButton.jsx", () => {
    Object.defineProperty(window, "matchMedia", windowMedia)
    const tree = renderer.create(<HelpButton onClick={jest.fn()} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Snapshot: ActionButtons.jsx", () => {
    Object.defineProperty(window, "matchMedia", windowMedia)
    const tree = renderer
      .create(
        <ActionButtons>
          <button type="submit">test button</button>
        </ActionButtons>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
