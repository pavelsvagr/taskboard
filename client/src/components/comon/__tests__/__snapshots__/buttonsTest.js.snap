// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Snapshot: ActionButtons.jsx 1`] = `
<div
  className="button-action"
  style={
    Object {
      "textAlign": "right",
    }
  }
>
  <div
    className=""
  >
    <button
      type="submit"
    >
      test button
    </button>
  </div>
</div>
`;

exports[`Buttons Snapshot: AddButton.jsx 1`] = `
<button
  className="ant-btn ant-btn-primary ant-btn-icon-only button"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="plus"
    className="anticon anticon-plus"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="plus"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <defs>
        <style />
      </defs>
      <path
        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
      />
      <path
        d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
      />
    </svg>
  </span>
</button>
`;

exports[`Buttons Snapshot: DragButton.jsx 1`] = `
<button
  className="ant-btn ant-btn-circle ant-btn-icon-only ant-btn-background-ghost"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="menu"
    className="anticon anticon-menu"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="menu"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</button>
`;

exports[`Buttons Snapshot: GoogleLoginButton.jsx 1`] = `
<a
  className="ant-btn ant-btn-primary ant-btn-lg login"
  href="/auth/google"
  onClick={[Function]}
>
  <span
    aria-label="google"
    className="anticon anticon-google"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="google"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M881 442.4H519.7v148.5h206.4c-8.9 48-35.9 88.6-76.6 115.8-34.4 23-78.3 36.6-129.9 36.6-99.9 0-184.4-67.5-214.6-158.2-7.6-23-12-47.6-12-72.9s4.4-49.9 12-72.9c30.3-90.6 114.8-158.1 214.7-158.1 56.3 0 106.8 19.4 146.6 57.4l110-110.1c-66.5-62-153.2-100-256.6-100-149.9 0-279.6 86-342.7 211.4-26 51.8-40.8 110.4-40.8 172.4S151 632.8 177 684.6C240.1 810 369.8 896 519.7 896c103.6 0 190.4-34.4 253.8-93 72.5-66.8 114.4-165.2 114.4-282.1 0-27.2-2.4-53.3-6.9-78.5z"
      />
    </svg>
  </span>
  <span>
    Login With Google
  </span>
</a>
`;

exports[`Buttons Snapshot: HelpButton.jsx 1`] = `
<button
  className="ant-btn ant-btn-dashed ant-btn-circle ant-btn-lg ant-btn-icon-only"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="question"
    className="anticon anticon-question"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="question"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M764 280.9c-14-30.6-33.9-58.1-59.3-81.6C653.1 151.4 584.6 125 512 125s-141.1 26.4-192.7 74.2c-25.4 23.6-45.3 51-59.3 81.7-14.6 32-22 65.9-22 100.9v27c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-27c0-99.5 88.6-180.4 197.6-180.4s197.6 80.9 197.6 180.4c0 40.8-14.5 79.2-42 111.2-27.2 31.7-65.6 54.4-108.1 64-24.3 5.5-46.2 19.2-61.7 38.8a110.85 110.85 0 00-23.9 68.6v31.4c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-31.4c0-15.7 10.9-29.5 26-32.9 58.4-13.2 111.4-44.7 149.3-88.7 19.1-22.3 34-47.1 44.3-74 10.7-27.9 16.1-57.2 16.1-87 0-35-7.4-69-22-100.9zM512 787c-30.9 0-56 25.1-56 56s25.1 56 56 56 56-25.1 56-56-25.1-56-56-56z"
      />
    </svg>
  </span>
</button>
`;
